library/network/src/lib/y2firewall/helpers/interfaces.rb

```diff
@@ -44,7 +44,7 @@ def firewalld
       #
       # @return [Array<String>] default zone interface names
       def default_interfaces
-        known_interfaces.select { |i| i["zone"].to_s.empty? }.map { |i| i["id"] }
+        known_interfaces.select { |i| !i.zone }.map(&:name)
       end
 
       # Return the zone name for a given interface from the firewalld instance
```

Comment:
the method `interface_zone` below could be simplified to `Y2Firewall::Firewalld::Interface.build(name).zone`

Reply:
What about using `#reject`?

```ruby
known_interfaces.reject(&:zone).map(&:name)
```